### PR TITLE
BUG: crosstab with margins=True and dropna=False 

### LINKED ISSUE #12642: BUG: crosstab with margins=True and dropna=False
xref #12614 
```
In [1]: df = pd.DataFrame({'a': [1, 2, 2, 2, 2, np.nan],'b': [3, 3, 4, 4, 4, 4]})

In [2]: pd.crosstab(df.a, df.b, margins=True, dropna=True)
Out[2]: 
b    3  4  All
a             
1.0  1  0    1
2.0  1  3    4
All  2  3    5

In [3]: pd.crosstab(df.a, df.b, margins=True, dropna=False)
KeyError: 'Level None not found'
```

### PR DESCRIPTION
**Pull request by OXPHOS**
- [+] closes #12642 
- [+] tests added / passed
- [+] passes `git diff upstream/master | flake8 --diff`
- [+] whatsnew entry

### CODE REVIEW

File: pandas/tools/tests/test_pivot.py [hunk 1]
```diff
@@ -936,6 +936,84 @@ def test_crosstab_no_overlap(self):
 
         tm.assert_frame_equal(actual, expected)
 
+    def test_margin_ignore_dropna_bug(self):
+        # GH 12577
```
Comment (jreback) [inline pandas/tools/tests/test_pivot.py]: so rename to ``test_margin_dropna``

Comment (OXPHOS) [inline pandas/tools/tests/test_pivot.py]: Trying to make sure I got this right -  

I current have two test methods respectively for this one and #12614. They're both `margins` and `dropna` issues for sure.

And I should merge these two methods into one, and name it `test_margin_dropna`, correct?

And I'll keep the issue number here?

**Comment by OXPHOS** [comment]
I'll wait for #12327

File: pandas/tools/tests/test_pivot.py (hunk 1 above)
Comment (jreback) [inline pandas/tools/tests/test_pivot.py]: yes that sounds fine (and just add this number after the other one). Idea is to have same tests together even though they occurred via different issues.

**Comment by jreback** [comment]
make sure you rebase on master (to pick up the merges from the other issue). 

you can submit a separate PR that does #12327 if you want (just incorporate those commits) and fixup as I have suggested.

**Comment by OXPHOS** [comment]
- Rebased.
- #12327 already solved the problem.
- The test file I generated may be useful
- But I don't know how to include commits from #12327 .  I assume I can't just copy his final codes. I tried forking from him but it didn't work like that.

**Comment by jreback** [comment]
ideally you would clone his branch
then cherry pick commits you want onto yours

**Pull request closed without merging**